### PR TITLE
[pre-depends] bump m4 to 1.4.17

### PR DESCRIPTION
**Pull request by Rechi**
- version bump fixes osx clang build issue
- fix LIBDYLIB definition that it not always reinstalls if nothing has changed

### CODE REVIEW

File: tools/depends/pre-depends/m4-pre-depends/Makefile [hunk 1]
```diff
@@ -1,19 +1,20 @@
 include ../../Makefile.include.in
```
Comment (yol) [inline tools/depends/pre-depends/m4-pre-depends/Makefile:1]: Just caught my eye, but: What good does including the `.in` file here?

Comment (Rechi) [reply tools/depends/pre-depends/m4-pre-depends/Makefile:1]: define at least `BASE_URL`

Comment (Rechi) [reply tools/depends/pre-depends/m4-pre-depends/Makefile:1]: ../../Makefile.include doesn't exist at the time this file is used

Comment (yol) [reply tools/depends/pre-depends/m4-pre-depends/Makefile:1]: Fair enough. Ensures lots of env variables with werid values but it doesn't seem to be a problem.

Comment (yol) [reply tools/depends/pre-depends/m4-pre-depends/Makefile:1]: > ../../Makefile.include doesn't exist at the time this file is used

Yes, best would be to separate the configured and the unconfigured (static) stuff into two files. Just nitpicking though ;-)

**Pull request merged**